yubikey-manager: 3.1.2 -> 4.0.1

veehaitch opened the pull request:
<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!
List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://nixos.org/manual/nixpkgs/unstable/#chap-reviewing-contributions
-->

###### Motivation for this change

Update YubiKey Manager CLI to v4.0.1.

###### Things done

- Updated Python package `fido2` to v0.9.1.
- Adapt Python package `aws-adf` to the new `fido2` version.
- Pinned `fido2` to v0.8.1 for Python package `solo-python` as it [does not support >=0.9.0](https://github.com/solokeys/solo-python/issues/110).
- Updated `yubikey-manager` to `v4.0.1`.

I tested `yubikey-manager` on macOS 11.2.3 with a YubiKey 5C.

The import check for `solo.cli` for package `solo-python` fails on macOS. This seems, however, unrelated to the changes of this PR as a `nix build -L nixpkgs#python3Packages.solo-python --rebuild` results in the same error. NixOS on x86_64 works fine.

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [x] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

veehaitch committed:
pythonPackages.fido2: 0.8.1 -> 0.9.1

veehaitch committed:
yubikey-manager: 3.1.2 -> 4.0.1

veehaitch committed:
python3Packages.aws-adfs: loosen fido2 version constraint

r-rmcgibbo commented:
Result of `nixpkgs-review pr 118725` at f62afcde run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>9 packages built successfully:</summary>
  <ul>
    <li>aws-adfs (python38Packages.aws-adfs)</li>
    <li>gen-oath-safe</li>
    <li>python38Packages.fido2</li>
    <li>python38Packages.solo-python</li>
    <li>python39Packages.aws-adfs</li>
    <li>python39Packages.fido2</li>
    <li>python39Packages.solo-python</li>
    <li>yubikey-manager</li>
    <li>yubikey-manager-qt</li>
  </ul>
</details>
<details>
  <summary>1 suggestion:</summary>
  <ul>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/python-include-tests.md">python-include-tests</a>

Consider adding a `checkPhase` for tests, or if not feasible, `pythonImportsCheck`.

Near pkgs/tools/misc/yubikey-manager/default.nix:50:0:
```
   |
50 |     description = "Command line tool for configuring any YubiKey over all USB transports";
   | ^
```
</li>
  </ul>
</details>

SuperSandro2000 commented:
@veehaitch please take a look at the bot suggestions.

drewrisinger commented on pkgs/development/python-modules/fido2/default.nix:
```diff
@@ -9,17 +9,27 @@
 
   propagatedBuildInputs = [ six cryptography ];
 
   checkInputs = [ mock pyfakefs ];
```
Maybe try using ``checkInputs = [ pytestCheckHook ... ];``? You might not need the custom checkPhase if that works. Pytest should nominally run all ``unittest`` tests out-of-the-box.

veehaitch replied:
Unfortunately, that approach doesn't seem to work.

drewrisinger commented on pkgs/development/python-modules/solo-python/default.nix:
```diff
@@ -11,7 +12,17 @@
-
+let
+  # https://github.com/solokeys/solo-python/issues/110#issuecomment-802945287
+  fido2_0_8_1 = fido2.overridePythonAttrs(oldAttrs: rec {
```
This is a red flag to me. Policy is to not override any python packages in the default package set, because that can cause package conflicts if e.g. someone tries to have a python environment with e.g. ``solo-python`` and ``fido2`` in the same environment.

If you're just overriding this for purposes of ``yubikey-manager``, you could try something along the lines of ``conan``: https://github.com/NixOS/nixpkgs/blob/17c66529f686d2138f7a5fdcc66f8cf6c7e052a5/pkgs/development/tools/build-managers/conan/default.nix.

Otherwise, maybe try unpinning the ``fido2`` dependency in ``solo-python``? Since the comment that you linked says that ``solo-python`` isn't being updated soon, maybe we just have to take the hit and mark ``solo-python`` as broken. @jonringer, thoughts?

veehaitch replied:
Thanks for your comment. In fact, I was expecting some hesitation regarding this change but I couldn't think of a proper solution on how to deal with this other than marking `solo-python` as broken. If this is acceptable, I'm happy to do that.

prusnak replied:
+1 for making `solo-python` broken for now with nice comment explaining where it should be marked as unbroken again

veehaitch replied:
Done! Thanks for reviewing this so quickly, @drewrisinger and @prusnak!

drewrisinger reviewed (changes requested):
* [ ] Diff needs changes, has an override in the default python package set.
* [x] Commits LGTM
* [x] Builds via ``nix-review`` on x86-64-linux:

```
10 packages built:
aws-adfs gen-oath-safe python38Packages.fido2 python38Packages.solo-python python39Packages.aws-adfs python39Packages.fido2 python39Packages.solo-python yubikey-manager yubikey-manager-qt yubioath-desktop
```

veehaitch committed:
pythonPackages.makefun: init at 1.11.2

veehaitch committed:
yubikey-manager: run tests

veehaitch committed:
python3Packages.solo-python: mark package as broken

solo-python currently does not support fido2 >= v0.9. Hence, we mark the
package as broken until upstream intoduces support. Starting point to
track this issue: https://github.com/solokeys/solo-python/issues/110.

prusnak commented:
Please drop commits 1f0519d583a5d8205e40ba2802b4b481d496d7eb and 10c62d83accce3e544d6a473409c52de207e6aa2 - no need to include a commit and its revert into the PR.

veehaitch commented:
> Please drop commits [1f0519d](https://github.com/NixOS/nixpkgs/commit/1f0519d583a5d8205e40ba2802b4b481d496d7eb) and [10c62d8](https://github.com/NixOS/nixpkgs/commit/10c62d83accce3e544d6a473409c52de207e6aa2) (via `git rebase -i master`) - no need to include a commit and its revert into the PR.

Sure, just included them for the sake of clarity of this PR.

SuperSandro2000 commented:
This is a semi-automatic executed nixpkgs-review with [nixpkgs-review-checks extension](https://github.com/SuperSandro2000/nixpkgs-review-checks). It is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).
If you have any questions or problems please reach out to SuperSandro2000 on IRC.

Result of `nixpkgs-review pr 118725` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
2 packages marked as broken and skipped:
</summary>
<ul>
<li>python38Packages.solo-python</li>
<li>python39Packages.solo-python</li>
</ul>
</details>
<details>
<summary>
10 packages built:
</summary>
<ul>
<li>aws-adfs (python38Packages.aws-adfs)</li>
<li>gen-oath-safe</li>
<li>python38Packages.fido2</li>
<li>python38Packages.makefun</li>
<li>python39Packages.aws-adfs</li>
<li>python39Packages.fido2</li>
<li>python39Packages.makefun</li>
<li>yubikey-manager</li>
<li>yubikey-manager-qt</li>
<li>yubioath-desktop</li>
</ul>
</details>

TredwellGit commented:
This breaks at least yubioath-desktop and yubikey-manager-qt on NixOS.
